store/internal/btree/btree.go

```diff
@@ -28,25 +28,36 @@ type BTree[V any] struct {
 
 // NewBTree creates a wrapper around `btree.BTreeG`.
 func NewBTree[V any]() BTree[V] {
-	return BTree[V]{
-		tree: btree.NewBTreeGOptions(byKeys[V], btree.Options{
-			Degree:  bTreeDegree,
-			NoLocks: false,
-		}),
+	return BTree[V]{}
+}
+
+func (bt *BTree[V]) init() {
+	if bt.tree != nil {
+		return
 	}
+	bt.tree = btree.NewBTreeGOptions(byKeys[V], btree.Options{
+		Degree:  bTreeDegree,
+		NoLocks: false,
+	})
 }
 
 // Set supports nil as value when used as overlay
-func (bt BTree[V]) Set(key []byte, value V) {
+func (bt *BTree[V]) Set(key []byte, value V) {
+	bt.init()
 	bt.tree.Set(newItem(key, value))
 }
 
 func (bt BTree[V]) Get(key []byte) (V, bool) {
+	if bt.tree == nil {
+		var zero V
+		return zero, false
+	}
 	i, found := bt.tree.Get(newItemWithKey[V](key))
 	return i.value, found
 }
 
-func (bt BTree[V]) Delete(key []byte) {
+func (bt *BTree[V]) Delete(key []byte) {
+	bt.init()
```

Comment:
do we just return if bt.tree is nil here

Reply:
good catch👍